pkg/process/util/address.go

```diff
@@ -9,28 +9,48 @@ import (
 	"encoding/binary"
 	"net"
 	"sync"
+
+	"inet.af/netaddr"
 )
 
 // Address is an IP abstraction that is family (v4/v6) agnostic
-type Address interface {
-	Bytes() []byte
-	WriteTo([]byte) int
-	String() string
-	IsLoopback() bool
-	Len() int
+type Address struct {
+	netaddr.IP
 }
 
-// AddressFromNetIP returns an Address from a provided net.IP
-func AddressFromNetIP(ip net.IP) Address {
-	if v4 := ip.To4(); v4 != nil {
-		var a v4Address
-		copy(a[:], v4)
-		return a
+// WriteTo writes the address byte representation into the supplied buffer
+func (a Address) WriteTo(b []byte) int {
+	if a.Is4() {
+		v := a.As4()
+		return copy(b, v[:])
+	}
+
+	v := a.As16()
+	return copy(b, v[:])
+
```

Comment:
```suggestion
	return copy(b, a.Bytes())
```

Reply:
AFAIR `Bytes()` allocates (the slice descriptor escapes to heap) which is what we're trying to avoid here.
I'll add benchmarks to the codebase

Reply:
```
BenchmarkBytes
BenchmarkBytes-4              	85095097	        13.30 ns/op	       4 B/op	       1 allocs/op
BenchmarkWriteTo
BenchmarkWriteTo-4            	290205878	         4.100 ns/op	       0 B/op	       0 allocs/op
```

Reply:
If you adjust `WriteTo` as suggested, does it still escape? Sometimes the compiler is smart.

Reply:
That's a good point! But doesn't work 😭 
```
BenchmarkBytes
BenchmarkBytes-4              	92096274	        13.63 ns/op	       4 B/op	       1 allocs/op
BenchmarkWriteTo
BenchmarkWriteTo-4            	86369061	        14.61 ns/op	       4 B/op	       1 allocs/op
```

Reply:
Le sad. :homerdisappear: